Adds Liquid Razor Extension Method & Markdown Helper parses liquid

deanmarcussen opened the pull request:
Suggested on gitter by @sebastienros and fixes https://github.com/OrchardCMS/OrchardCore/issues/3532

Two extensions, one just to render liquid as `IHtmlContent` and the other to render liquid as a raw string that can be passed to another extension method like the markdown helper.

Plus some docs

sebastienros commented on src/OrchardCore.Modules/OrchardCore.Liquid/Razor/LiquidRazorHelperExtensions.cs:
```diff
@@ -0,0 +1,74 @@
+    /// <param name="liquid">The liquid to parse.</param>
+    /// <param name="model">A model to bind against.</param>
+    /// <summary>
+    public static async Task<IHtmlContent> LiquidToHtml(this IOrchardHelper orchardHelper, string liquid, object model)
```
I don't think we need the Raw one. Also this one should specify the HtmlEncoder as in razor we will mostly render html, and the returned `IHtmlContent` assumes that this is already encoded.

deanmarcussen replied:
So I added the Raw one to make it work for https://github.com/OrchardCMS/OrchardCore/issues/3532 so that it could give a normal string back, so that you could then send that string to the markdown extension (which takes a string) `IHtmlContent.ToString()` was giving encoding back and the markdown helper wasn't rendering it correctly. Maybe there is another way for that? 
I will update to specify the HtmlEncoder

deanmarcussen replied:
Hmm, so could just update the markdown helper to be `async` and always render liquid with it, instead of the Raw one, if you prefer?

deanmarcussen commented:
Removed Raw helper, and updated Markdown helper to be `MarkdownToHtmlAsync` and parse liquid.

For encoding the `ILiquidTemplateManager.RenderAsync` uses `HtmlEncoder.Default` so this should be coming back Html encoded.

**Note** I haven't provided a backwards compatible `MarkdownToHtml` extension method. So little bit of breaking change in this. Can do a backwards compatible one if wanted?
